tests/transforms/test_optimization/test_cancel_inverses.py

```diff
@@ -329,3 +329,343 @@ def test_cancel_inverses_jax(self):
         # Check operation list
         ops = transformed_qnode.qtape.operations
         compare_operation_lists(ops, expected_op_list, expected_wires_list)
+
+
+class TestCancelInversesNew:
```

Comment:
While all these integration tests are necessary, I also 100% want to make sure we have unit tests as well.